### PR TITLE
chore: Increase parallelism in Test Suite

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Increases parallelism in Test Suite to make test execution faster.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: .github/workflows/test-suite.yml [hunk 1]
```diff
@@ -52,10 +52,10 @@ jobs:
       provider_version: ${{ matrix.provider_version }}
 
   acc-tests:
-    needs: [mig-tests, versions]
+    needs: [clean-before, versions]
```
Comment (oarbusi) [inline .github/workflows/test-suite.yml:55]: why the change from `mig-tests` to `clean-before`? I see that you added `mig-tests` in `clean-after` but not in the `clean-before`

Comment (lantoli) [reply .github/workflows/test-suite.yml:55]: so mig and acc run in parallel instead of being a chain of mig then acc

**Comment by maastha** [comment]
[q] just for my knowledge, how do we test this?

**Comment by lantoli** [comment]
@maastha I ran manually the Test Suite GH action in this branch.   for this change, i didn't expect until the end, i saw that both tests were running in parallel and seemed right and canceled.   for more complex changes i wait until the GH action in this branch finishes completely

**Review by AgustinBettati — APPROVED** [review]
LGTM, only risk I see is that organization reaches limit of projects but we can always revert and look into other options in that case.

**Comment by lantoli** [comment]
@AgustinBettati i don't think we reach org limit but could happen, however i think it will probably fail in other places.  anyway i'll wait until Mon so it doesn't interfere with TF 1.7 in case something fails

File: .github/workflows/test-suite.yml (hunk 1 above)
Comment (lantoli) [reply .github/workflows/test-suite.yml:55]: mig_test already had:

 mig-tests:
    needs: [clean-before, versions]

**Comment by lantoli** [comment]
@AgustinBettati it could happen but i guess it'll fail because other things.  so to go a bit more smoother I'm going to have mig and acc tests in paralel, but each on in serial and see what happens. if it works we can increase parallelism inside acc and mig tests
